### ISSUE TITLE
unarchive + sudo + Solaris = broken

### ISSUE THREAD

**Issue by ericcberg** [ansible/ansible#11282]
I am using ansible to run CIS CAT scripts and found that my script was running fine on my Linux hosts but on Solaris I was getting this output: 

fatal: [coeusdev.acsu.buffalo.edu] => ssh connection closed waiting for a privilege escalation password prompt

When running this step:

```
  - name: install the app
    unarchive: src=./ciscat.zip dest={{ install_dir }}
    when: not dir.stat.exists 
    sudo: yes
```

I tried manually copying the file over, to simplify matters, and ran this pared down script:

```
- hosts: solaris-dev
  sudo: yes
  gather_facts: no

  tasks:
  - name: install the app
    unarchive: copy=no src=./ciscat.zip dest=/opt/cis-cat-full
```

I got the same results. For some reason, it seems unarchive doesn't play nice with sudo through ansible with Solaris 10 target machines.

**Comment by bcoca** [comment]
what version of ansible? python?  verbose errors? (its faster if you just fill the template)

**Comment by ericcberg** [comment]
Sorry, didn't know there was a template. First time here.

Ansible 1.9.1
Python 2.7.8

Verbose errors:
PLAY [coeusdev] **************************************************************\* 

TASK: [install the app] ******************************************************\* 
<coeusdev.acsu.buffalo.edu> ESTABLISH CONNECTION FOR USER: eric
<coeusdev.acsu.buffalo.edu> EXEC ssh -C -tt -v -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/home/eric/.ansible/cp/ansible-ssh-%h-%p-%r" -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 coeusdev.acsu.buffalo.edu /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1434468083.39-207590490422431 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1434468083.39-207590490422431 && echo $HOME/.ansible/tmp/ansible-tmp-1434468083.39-207590490422431'
<coeusdev.acsu.buffalo.edu> EXEC ssh -C -tt -v -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/home/eric/.ansible/cp/ansible-ssh-%h-%p-%r" -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 coeusdev.acsu.buffalo.edu /bin/sh -c 'sudo -k && sudo -H -S -p "[sudo via ansible, key=vxvavrmjzydciomlubgjzmirhikubrni] password: " -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-vxvavrmjzydciomlubgjzmirhikubrni; rc=flag; [ -r /opt/cis-cat-full ] || rc=2; [ -f /opt/cis-cat-full ] || rc=1; [ -d /opt/cis-cat-full ] && rc=3; python -V 2>/dev/null || rc=4; [ x"$rc" != "xflag" ] && echo "${rc} "/opt/cis-cat-full && exit 0; (python -c '"'"'"'"'"'"'"'"'import hashlib; BLOCKSIZE = 65536; hasher = hashlib.sha1(); afile = open("'"'"'"'"'"'"'"'"'/opt/cis-cat-full'"'"'"'"'"'"'"'"'", "rb") buf = afile.read(BLOCKSIZE) while len(buf) > 0: hasher.update(buf) buf = afile.read(BLOCKSIZE) afile.close() print(hasher.hexdigest())'"'"'"'"'"'"'"'"' 2>/dev/null) || (python -c '"'"'"'"'"'"'"'"'import sha; BLOCKSIZE = 65536; hasher = sha.sha(); afile = open("'"'"'"'"'"'"'"'"'/opt/cis-cat-full'"'"'"'"'"'"'"'"'", "rb") buf = afile.read(BLOCKSIZE) while len(buf) > 0: hasher.update(buf) buf = afile.read(BLOCKSIZE) afile.close() print(hasher.hexdigest())'"'"'"'"'"'"'"'"' 2>/dev/null) || (echo '"'"'"'"'"'"'"'"'0 '"'"'"'"'"'"'"'"'/opt/cis-cat-full)'"'"''
fatal: [coeusdev.acsu.buffalo.edu] => ssh connection closed waiting for a privilege escalation password prompt

FATAL: all hosts have already failed -- aborting

PLAY RECAP *******************************************************************\* 
           to retry, use: --limit @/home/eric/test.retry

coeusdev.acsu.buffalo.edu  : ok=0    changed=0    unreachable=1    failed=0

**Comment by bcoca** [comment]
can you show me the command line used to execute this?

**Comment by bcoca** [comment]
FYI https://github.com/ansible/ansible/blob/devel/ISSUE_TEMPLATE.md (it is linked from the contribution guidelines, which are linked when you open new ticket).

**Comment by ericcberg** [comment]
ansible-playbook -K -vvv test.yml

Thanks. Sorry, I am doing several things in parallel and was putting in the ticket between jobs so I wasn't reading closely. :)

**Comment by bcoca** [comment]
can you try adding '-c paramiko' to the command line?

**Comment by ericcberg** [comment]
[eric@bishop ciscat]$ ansible-playbook -c paramiko -K -vvv test.yml 
SUDO password: 

PLAY [coeusdev] **************************************************************\* 

TASK: [install the app] ******************************************************\* 
<coeusdev.acsu.buffalo.edu> ESTABLISH CONNECTION FOR USER: eric on PORT 22 TO coeusdev.acsu.buffalo.edu
<coeusdev.acsu.buffalo.edu> EXEC /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1434468576.09-14278608207670 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1434468576.09-14278608207670 && echo $HOME/.ansible/tmp/ansible-tmp-1434468576.09-14278608207670'
<coeusdev.acsu.buffalo.edu> EXEC /bin/sh -c 'sudo -k && sudo -H -S -p "[sudo via ansible, key=qylqzdvybqgletxjxraajjxlekdlbliz] password: " -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-qylqzdvybqgletxjxraajjxlekdlbliz; rc=flag; [ -r /opt/cis-cat-full ] || rc=2; [ -f /opt/cis-cat-full ] || rc=1; [ -d /opt/cis-cat-full ] && rc=3; python -V 2>/dev/null || rc=4; [ x"$rc" != "xflag" ] && echo "${rc} "/opt/cis-cat-full && exit 0; (python -c '"'"'"'"'"'"'"'"'import hashlib; BLOCKSIZE = 65536; hasher = hashlib.sha1(); afile = open("'"'"'"'"'"'"'"'"'/opt/cis-cat-full'"'"'"'"'"'"'"'"'", "rb") buf = afile.read(BLOCKSIZE) while len(buf) > 0: hasher.update(buf) buf = afile.read(BLOCKSIZE) afile.close() print(hasher.hexdigest())'"'"'"'"'"'"'"'"' 2>/dev/null) || (python -c '"'"'"'"'"'"'"'"'import sha; BLOCKSIZE = 65536; hasher = sha.sha(); afile = open("'"'"'"'"'"'"'"'"'/opt/cis-cat-full'"'"'"'"'"'"'"'"'", "rb") buf = afile.read(BLOCKSIZE) while len(buf) > 0: hasher.update(buf) buf = afile.read(BLOCKSIZE) afile.close() print(hasher.hexdigest())'"'"'"'"'"'"'"'"' 2>/dev/null) || (echo '"'"'"'"'"'"'"'"'0 '"'"'"'"'"'"'"'"'/opt/cis-cat-full)'"'"''
fatal: [coeusdev.acsu.buffalo.edu] => ssh connection closed waiting for password prompt

FATAL: all hosts have already failed -- aborting

PLAY RECAP *******************************************************************\* 
           to retry, use: --limit @/home/eric/test.retry

coeusdev.acsu.buffalo.edu  : ok=0    changed=0    unreachable=1    failed=0

**Comment by bcoca** [comment]
after you login normally , does sudo prompt you for a password?

**Comment by ericcberg** [comment]
Here's the rest of the stuff from the template I missed:

Ansible Configuration: Only changed forks from 5 to 10.
Environment: Running ansible on Fedora 21, managing Solaris 10 non-global zones.

**Comment by ericcberg** [comment]
Yes.

**Comment by bcoca** [comment]
I cannot reproduce the issue when using against openindiana (my 'solaris test platform'), not sure if the differences with actual solaris are big enough. I would ping the mailing list/IRC to see if other solaris users are having the same issue

**Comment by ericcberg** [comment]
Will do.

**Comment by jimi-c** [comment]
Hi @ericcberg just following up on this issue, as it's been a month without any further comments. Are you still seeing this error?

Thanks!

**Comment by ericcberg** [comment]
Still seeing it. I haven't really done anything to try and troubleshoot further. It was holding up work, so I just copied and uncompressed the package in question using CSSH instead.

**Comment by jimi-c** [comment]
@ericcberg any chance you can test this on the devel branch to see if we've resolved it already? Parts of these action plugins were rewritten and in many cases some long standing bugs were fixed.

**Comment by ericcberg** [comment]
James Cammarata wrote on 07/17/2015 03:39 PM:

> @ericcberg https://github.com/ericcberg any chance you can test this
> on the devel branch to see if we've resolved it already? Parts of these
> action plugins were rewritten and in many cases some long standing bugs
> were fixed.

I pulled the latest build and it does, in fact, fix the issue. Thanks 
for your help!

## 

Eric Christian Berg
Senior Systems Administrator
Enterprise Infrastructure Services
State University of New York at Buffalo
716-645-5476

**Comment by bcoca** [comment]
closing as fixed in devel
